Feat: CRUD frontend connection

maxlshk opened the pull request:
- Implementation of CRUD operations on frontend, connected to backend. 
- Adaptability for all kinds of displays using `useMediaQuery` hook.
- Implementation of basic routes, no private routes yet.
- Inputs and forms separated into shared components. 
- Basic styling with Emotion.css.
- Added CORS on backend.

malyshkoserhii commented on packages/frontend/src/modules/app/components/TodoItem/TodoItem.tsx:
```diff
@@ -0,0 +1,77 @@
+	const handleDelete = (): Promise<void> => deleteTodo(todo.id);
+
+	const handleEdit = (): void => {
+		navigator(`/todos/edit/${todo.id}`);
```
ROUTER_KEYS

malyshkoserhii commented on packages/frontend/src/modules/app/components/TodoList/TodoList.tsx:
```diff
@@ -0,0 +1,82 @@
+		);
+	}
+	const filteredTodos = todos.filter((todo) => {
+		if (filter === 'ALL') {
```
use enum 'ALL', 'PRIVATE' etc.

replace lots of if tatements with object with dynamic keys

malyshkoserhii commented on packages/frontend/src/modules/app/pages/EditTodoPage/EditTodoPage.tsx:
```diff
@@ -0,0 +1,63 @@
+		console.log('edit data', data);
+		updateTodo(Number(id), data);
+		reset();
+		navigate('/todos');
```
ROUTER_KEYS

malyshkoserhii commented on packages/frontend/src/modules/app/pages/EditTodoPage/EditTodoPage.tsx:
```diff
@@ -0,0 +1,63 @@
+				defaultValue={todo.title}
+				type="input"
+				required
+			/>
```
add validation on FE side

malyshkoserhii commented on packages/frontend/src/shared/services/http.ts:
```diff
@@ -0,0 +1,28 @@
+import axios from 'axios';
+import { Todo } from '../services/types';
+
+const API_URL = 'http://localhost:3030/api';
```
.env

malyshkoserhii commented on packages/frontend/src/shared/services/http.ts:
```diff
@@ -0,0 +1,28 @@
+
+const API_URL = 'http://localhost:3030/api';
+
+export const getTodos = async (): Promise<Todo[]> => {
```
create HttpService. Please see company codestyle

maxlshk commented:
Fix: all pointed out issues reviewed and fixed ✓

- Used ROUTER_KEYS instead of navigation routes as strings.
- Replaced if-statements by using enum functionality.
- Added validation for form inputs.
- Moved necessary variables to .env file.
- Created `http.service.ts` as generic service class for CRUD operations and `todo.service.ts` file extending it's functionality.

Additionally:

- Added todo creation/editing form submission error status to handle potential unsuccessful operations.

malyshkoserhii commented on packages/frontend/src/shared/components/textinput/textinput.component.tsx:
```diff
@@ -21,17 +28,41 @@ export const TextInput: React.FC<FormFieldProps> = ({
-
-	return <input type={inputType} {...sharedProps} />;
+	return (
+		<div>
+			{type === 'textarea' ? (
+				<textarea {...sharedProps} />
+			) : (
+				<input type={inputType} {...sharedProps} />
+			)}
+			{error && (
+				<span className={errorMessageStyles}>{error.message}</span>
+			)}
+		</div>
```
split on two components: input and textarea
